src/components/TextField/TextField.scss

```diff
@@ -193,6 +200,21 @@ $stacking-order: (
   padding-bottom: spacing(tight);
 }
 
+.ClearButton {
+  z-index: z-index(contents, $stacking-order);
+  appearance: none;
+  background: none;
+  border: none;
+  padding: 0;
+  margin: 0 spacing(tight);
+  cursor: pointer;
```

Comment:
We have a mixin called `unstyled-button` that would be perfect for handling most of these styles.